chain/src/pipe.rs

```diff
@@ -260,19 +260,10 @@ pub fn process_block_header(header: &BlockHeader, ctx: &mut BlockContext<'_>) ->
 		header.height,
 	); // keep this
 
-	check_header_known(header, ctx)?;
-	validate_header(header, ctx)?;
-	Ok(())
-}
+	// Check if this header is already "known" from processing a previous block.
+	check_known(header, ctx)?;
 
-/// Quick in-memory check to fast-reject any block header we've already handled
-/// recently. Keeps duplicates from the network in check.
-/// ctx here is specific to the header_head (tip of the header chain)
-fn check_header_known(header: &BlockHeader, ctx: &mut BlockContext<'_>) -> Result<(), Error> {
-	let header_head = ctx.batch.header_head()?;
```

Comment:
It said `ctx here is specific to the header_head (tip of the header chain)`.

but in `check_know()`, the `check_known_head(header, ctx)` is using `ctx.batch.head()`, not `header_head`.

Reply:
This code is deleted.

Reply:
I mean your new code has very different behavior from the old `check_header_known` which use header_head.  now you're using `ctx.batch.head()`.

Reply:
We used to check for the header itself was known based on the header chain.
The only time we would know about headers and not yet now about the full blocks is during initial sync.
So it's safe to check if we know about the header based on the full chain - we're processing "header first" in order to process full blocks.